R/eatModel.r

```diff
@@ -1,55 +1,85 @@
+### multiple other functions which couldn't be usefully grouped.
 
-### called by .substituteSigns() and getConquestAdditionalTerms () -------------
+### called by defineModel/.substituteSigns() and getResults/getConquestAdditionalTerms()
 
-isLetter <- function ( string ) {
+isLetter <- function(string){
+  # checks
+  checkmate::assert_character(string)
+
+  # function
   splt <- strsplit(string, "")
-  isL  <- lapply(splt, FUN = function ( x ) {
-    ind <- which ( x %in% c( letters , LETTERS ))
-    x[setdiff(1:length(x),ind)] <- " "
-    x <- eatTools::crop(paste(x, sep="", collapse=""))
-    x <- unlist ( strsplit(x, " +") )
-    return(x)  } )
-  return(isL)}
+  isL <- lapply(splt, FUN = function(x) {
+    ind <- which(x %in% c(letters, LETTERS))
+    x[setdiff(1:length(x), ind)] <- " "
+    x <- eatTools::crop(paste(x, sep = "", collapse = ""))
+    x <- unlist(strsplit(x, " +"))
+    return(x)
+  })
+  return(isL)
+}
 
 ### not called ----------------------------------------------------------------------------
 
-simEquiTable <- function ( anchor, mRef, sdRef, addConst = 500, multConst = 100, cutScores) {
-                anchor<- eatTools::makeDataFrame(anchor)
-                if ( ncol(anchor) != 2) {
-                     warning(paste0("'anchor' has ",ncol(anchor)," columns. First column is used as item ID, second column is used as item parameter."))
-                }
-                if(!inherits(anchor[,2], c("integer", "numeric"))) {stop("Item parameter column must be numeric.")}
-                if(length(unique(anchor[,1])) != nrow(anchor)) {stop("Item ID column has duplicated entries.")}
-                dtmp  <- data.frame(rbind(1*(lower.tri(matrix(1, nrow = nrow(anchor), ncol = nrow(anchor)))),1))
-                dtmp  <- data.frame(dtmp, score = rowSums(dtmp) , irtoys::wle(dtmp, cbind(1, anchor[,2], 0)), stringsAsFactors = FALSE)
-                dtmp[,"bista"] <- (dtmp[,"est"] - mRef) / sdRef * multConst + addConst
-                dtmp[,"ks"]    <- eatTools::num.to.cat ( x = dtmp[,"bista"], cut.points = cutScores[["values"]], cat.values = cutScores[["labels"]])
-                dig   <- 0
-                while ( length(which(round(dtmp[,"bista"], digits = dig) %in% cutScores[["values"]])) > 0) {dig <- dig + 1}
-                shrt  <- do.call("rbind", by ( data = dtmp, INDICES = dtmp[,"ks"], FUN = function ( sks ) { data.frame ( score = paste(c(min(sks[,"score"]), max(sks[,"score"])), collapse=" bis "), estimate = paste(round(c(min(sks[,"est"]), max(sks[,"est"])),digits=2), collapse=" bis "), bista = paste(round(c(min(sks[,"bista"]), max(sks[,"bista"])),digits=dig), collapse=" bis "), ks=unique(sks[,"ks"]), stringsAsFactors=FALSE)}))
-                return(list ( complete = dtmp[,c("score", "est", "bista", "ks")], short = shrt))}
+simEquiTable <- function(anchor, mRef, sdRef, addConst = 500, mulConst = 100, cutScores){
+  # checks/prep
+  anchor <- eatTools::makeDataFrame(anchor)
+  if(ncol(anchor) != 2){

```

Comment:
I left these as they are. They are more detailed than checkmate, I think

Reply:
All right, thanks! You could consider adding checks for arguments `mRef`, `sdRef`, `addConst`, `mulConst` and `cutScores`. `mRef`, `sdRef`, `addConst`, and `mulConst` should be numeric (length = 1), `sdRef`, `addConst`, and `mulConst` additionally must be positive. `cutScores` must be a list with a predefined structure. This is potentially hard to check with checkmate.

Reply:
depending on what kind of structure. some tests are definitely possible: 
![grafik](https://github.com/user-attachments/assets/52535145-ffd9-454a-8344-102996fcdbc5)

Reply:
Ah, I understand. See the help page `?simEquiTable` what structure is requested for the `cutScores` argument. The `values` element must contain the cut scores in ascending order. The `labels` argument is optional and should contain labels (i.e., 1, 2, 3, 4, ...) which give the denomination of the competence levels. In the english domain, labels are denoted A1.1, A1.2, for example. 
Which makes things even more complicated: The `transformToBista` function has an argument `cuts` that is similar to `cutScores`, but not identical. `cuts` additionally must include the domain name. 
```r
# valid values for cutScores (simEquiTable function)
list ( values = 390+0:3*75, labels = c("I", "II", "III", "IV", "V") )
list ( values = 390+0:3*75)
# exemplary valid values for cutScores (transformToBista function)
list ( domainreading = list ( values = 390+0:3*75),
       domainlistening = list ( values = 360+0:3*85, labels = c("A1", "A2", "B1", "B2", "C1"))
```
As I understand `assert_list`, some but not all requirements of `cutScores` and `cuts` can be checked.

Reply:
When cutScores only has the `values` argument without `labels`, does it still have to be a list?